content/en/blog/_posts/2021-03-30-metrics-stability-ga/2021-03-30-metrics-stability-hits-ga.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: blog
+title: 'Kubernetes 1.21: Metrics Stability hits GA'
+date: 2021-03-30
+slug: kubernetes-release-1.21-metrics-stability-ga
+---
+
+**Authors**: Han Kang (Google). Elana Hashman (Red Hat)
+
+Kubernetes 1.21 marks the graduation of the metrics stability framework and along with it, the first officially supported stable metrics. Not only do stable metrics come with supportability guarantees, the metrics stability framework brings escape hatches that you can use if you encounter problematic metrics.
+
+### What are stable metrics and why do we need them?
+A stable metric is one which, from a consumption point of view, can be reliably consumed across a number of Kubernetes versions without risk of ingestion failure.
+
+Metrics stability is an ongoing community concern. Cluster monitoring infrastructure often assumes the stability of some control plane metrics, so we have introduced a mechanism for versioning metrics as a proper API, with stability guarantees around a formal metrics deprecation process.
+
+### What are the stability levels for metrics?
+
+Metrics can currently have one of two stability levels: alpha or stable.
+
+_Alpha metrics_ have no stability guarantees; as such they can be modified or deleted at any time. At this time, all Kubernetes metrics implicitly fall into this category.
+
+_Stable metrics_ can be guaranteed to not change, except that the metric may become marked deprecated for a future Kubernetes version. By not change, we mean three things:
+
+1. the metric itself will not be deleted or renamed
+2. the type of metric will not be modified
+3. no labels can be added or removed from this metric
+
+From an ingestion point of view, it is backwards-compatible to add or remove possible values for labels which already do exist, but not labels themselves. Therefore, adding or removing values from an existing label is permitted. Stable metrics can also be marked as deprecated for a future Kubernetes version, since this is tracked in a metadata field and does not actually change the metric itself.
+
+Removing or adding labels from stable metrics is not permitted. In order to add or remove a label from an existing stable metric, one would have to introduce a new metric and deprecate the stable one; otherwise this would violate compatibility agreements.
```

Comment:
Should we add a link to the current list of stable metrics?

Reply:
do you have a link to the auto-generated list?

Reply:
Ref: https://github.com/kubernetes/website/issues/24915

It looks like the link was removed.

Reply:
Feel free to put it back - ideally before the blog post goes out!

Reply:
Added a link :)